internal/cephfs/core/filesystem.go

```diff
@@ -25,76 +25,99 @@ import (
 	"github.com/ceph/ceph-csi/internal/util/log"
 )
 
-func (vo *VolumeOptions) getFscID(ctx context.Context) (int64, error) {
-	fsa, err := vo.conn.GetFSAdmin()
+// FileSystem is the interface that holds the signature of filesystem methods
+// that interacts with CephFS filesystem API's.
+type FileSystem interface {
+	GetFscID(context.Context, string) (int64, error)
+	GetMetadataPool(context.Context, string) (string, error)
+	GetFsName(context.Context, int64) (string, error)
+}
+
+// fileSystem is the implementation of FileSystem interface.
+type fileSystem struct {
+	conn *util.ClusterConnection
+}
+
+// NewFileSystem returns a new instance of fileSystem.
+func NewFileSystem(conn *util.ClusterConnection) FileSystem {
+	return &fileSystem{
+		conn: conn,
+	}
+}
+
+// GetFscID returns the ID of the filesystem with the given name.
```

Comment:
The documentation for functions from an interface can be placed at the interface. In case there are multiple implementation of the interface, there is no need to repeat the generic description.

Details for the implementation specifics of this function should of course be documented here, instead of at the interface level.

Reply:
Done.